plugin/storage/es/samplingstore/storage.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samplingstore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/olivere/elastic"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/model"
+	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/plugin/storage/es/samplingstore/dbmodel"
+)
+
+const (
+	samplingIndex        = "jaeger-sampling-"
+	throughputType       = "throughput-sampling"
+	probabilitiesType    = "probabilities-sampling"
+	indexPrefixSeparator = "-"
+)
+
+type SamplingStore struct {
+	client              func() es.Client
+	logger              *zap.Logger
+	samplingIndexPrefix string
+	indexDateLayout     string
+	maxDocCount         int
+}
+
+type SamplingStoreParams struct {
+	Client          func() es.Client
+	Logger          *zap.Logger
+	IndexPrefix     string
+	IndexDateLayout string
+	MaxDocCount     int
+}
+
+func NewSamplingStore(p SamplingStoreParams) *SamplingStore {
+	return &SamplingStore{
+		client:              p.Client,
+		logger:              p.Logger,
+		samplingIndexPrefix: prefixIndexName(p.IndexPrefix, samplingIndex),
+		indexDateLayout:     p.IndexDateLayout,
+		maxDocCount:         p.MaxDocCount,
+	}
+}
+
+func (s *SamplingStore) InsertThroughput(throughput []*model.Throughput) error {
+	ts := time.Now()
+	writeIndexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, ts)
+	s.writeThroughput(writeIndexName, ts, throughput)
+	return nil
+}
+
+func (s *SamplingStore) InsertProbabilitiesAndQPS(hostname string,
+	probabilities model.ServiceOperationProbabilities,
+	qps model.ServiceOperationQPS,
+) error {
+	ts := time.Now()
+	writeIndexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, ts)
+	val := dbmodel.ProbabilitiesAndQPS{
+		Hostname:      hostname,
+		Probabilities: probabilities,
+		QPS:           qps,
+	}
+	s.writeProbabilitiesAndQPS(writeIndexName, ts, val)
+	return nil
+}
+
+func (s *SamplingStore) writeProbabilitiesAndQPS(indexName string, ts time.Time, pandqps dbmodel.ProbabilitiesAndQPS) {
+	s.client().Index().Index(indexName).Type(probabilitiesType).
+		BodyJson(&dbmodel.TimeProbabilitiesAndQPS{
```

Comment:
data conversion it not necessary again

Reply:
It is necessary because in this type conversion(`dbmodel.ProbabilitiesAndQPS` -> `dbmodel.TimeProbabilitiesAndQPS`), we are adding a timestamp also, which is important for it to work properly.